### PR TITLE
(#1581) Card Storybook Update

### LINKED ISSUE #1581: Update storybook structure for nci-card
### `nci-card` storybook structure
1. Default (folder)
    1. Default
    2. Card Group
2. Test cases (folder)
    1. Card in Text, Left
    2. Card in Text, Right
    3. Card Group without Description (removed on one or more cards)

Other tasks:
* Clean up all of the regressions (redundant breakpoint scenarios) associated with this component
* Create ncids-twig template
* Make sure scenarios still reflect all of the test cases that we have, but we do not need them as examples in the storybook structure
* 
> **ESTIMATE** TBD

### Additional details / screenshot
- [Storybook structure wiki](https://github.com/NCIOCPL/ncids/wiki/Storybook-Structure)

### Related Tickets
* Parent ticket: #1224

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #1581

### CODE REVIEW

**Review by andyvanavery31 — CHANGES_REQUESTED** [review]
Could you please change the icon of Default, and Test Cases to have a folder instead of the 2x2 grid? Could you also please add a space in "TestCases"?

For regression image redundancies:

- ncids_nci_card_active_0_document_0_widescreen.png and ncids_nci-card_focus_0_document_0_widescreen.png

**Comment by karlikpj** [comment]
The Icon's are via storybook and how the paths/folders are created. The end sections will always be a grid currently.

**Review by andyvanavery31 — APPROVED** [review]
Looks good to me!

File: packages/ncids-twig/components/nci-card/nci-card.twig [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<div id="nci-card" class="grid-container">
+	{% if float %}
+		<div class="usa-prose">
```
Comment (olitharp-nci) [inline packages/ncids-twig/components/nci-card/nci-card.twig:3]: The twig file should **only** be the card itself. All the content that belongs to the story to help with testing needs to live in the storybook files.

File: testing/ncids-css-testing/stories/components/nci-card/nci-card.scenarios.js [hunk 2]
```diff
@@ -106,48 +91,64 @@ module.exports = [
...
 		],
 	},
 	/* PostInteractWait Accounts for Image Zoom Transition */
 	{
-		label: 'nci-card hover',
-		storyId: 'components-nci-card--default',
+		label: 'nci-card default hover',
+		storyId: 'components-card-default--default',
 		hoverSelector: '.nci-card a',
 		postInteractionWait: 500,
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/components/nci-card/nci-card.scenarios.js:122]: can you change `postInteractionWait` to `readySelector` on all of these that have it. An arbitrary number will slow down the tests

File: testing/ncids-css-testing/stories/components/nci-card/nci-card.testcases.group.without.description.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/components/nci-card/nci-card.testcases.group.without.description.stories.jsx:1]: Merge all test cases to one story file. Get rid of float. You can create a twig file here that calls the ncids-twig package. The params that are being passed to the twig should only be things that are actually a part of the card component

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: packages/ncids-twig/components/nci-card/nci-card.twig [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<div class="nci-card tablet-lg:grid-col-4 {{ modifier }} ">
+	<a href="#" aria-label="Card">
```
Comment (olitharp-nci) [inline packages/ncids-twig/components/nci-card/nci-card.twig:2]: links labels should not be hardcoded either

File: testing/ncids-css-testing/.backstop/reference/ncids_nci-card_default_0_document_3_desktop.png
Comment (olitharp-nci) [inline testing/ncids-css-testing/.backstop/reference/ncids_nci-card_default_0_document_3_desktop.png:1]: were the margins supposed to change on these or did some classes get moved to the wrong element?

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: packages/ncids-twig/components/nci-card/nci-card-group.twig [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<div class="grid-container">
+    <ul class="nci-card-group">
+        {% for item in items %}
+            <li class="nci-card tablet-lg:grid-col-4">
+                <a href="#" aria-label="{{ ariaLabel }}">
```
Comment (olitharp-nci) [inline packages/ncids-twig/components/nci-card/nci-card-group.twig:5]: don't hard code href

File: packages/ncids-twig/components/nci-card/nci-card-group.twig [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+                    <picture class="nci-card__image">
+                        <source
+                        media="(min-width: 880px)"
+                        srcset="{{ item.imageSource }}"
+                        />
+                        <img src="{{ item.imageSrc }}" />
```
Comment (olitharp-nci) [inline packages/ncids-twig/components/nci-card/nci-card-group.twig:11]: add missing alts

File: packages/ncids-twig/components/nci-card/nci-card.twig [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<div id="nci-card" class="grid-container">
+	<div class="nci-card tablet-lg:grid-col-4 {{ modifier }} ">
+		<a href="#" aria-label="{{ ariaLabel }}">
```
Comment (olitharp-nci) [inline packages/ncids-twig/components/nci-card/nci-card.twig:3]: hardcoded href

File: packages/ncids-twig/components/nci-card/nci-card.twig [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+			<picture class="nci-card__image">
+				<source media="(min-width: 880px)" srcset="{{ imageSource }}"/>
+				<img src="{{ imageSrc }}"/>
```
Comment (olitharp-nci) [inline packages/ncids-twig/components/nci-card/nci-card.twig:6]: missing alt

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: testing/ncids-css-testing/stories/components/nci-card/nci-card.group.default.stories.jsx
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/components/nci-card/nci-card.group.default.stories.jsx:1]: This should be exported in testing/ncids-css-testing/stories/components/nci-card/nci-card.default.stories.jsx

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Comment by andyvanavery31** [comment]
@karlikpj I'm seeing something weird happening with the Test Case of Card Group Without Description, the first card is shrinking the height of the image so it is shorter than the rest, but this is not happening with the Default > Card Group. Also, this is not showing up on the regression images. The regression images look good as long as they are aligned with whatever this fix is.

**Comment by andyvanavery31** [comment]
@karlikpj can you please return the description to the default card? I believe that should be the default. Additionally, I am seeing the difference between the left-most card and the rest of the cards got even more obvious when looking at Card Group (works correctly) vs Test Cases > Card Group Without Description (first card's image got really small and the spacing between the left-most and middle card increased)

**Review by andyvanavery31 — APPROVED** [review]
Looks good to me!

**Review by olitharp-nci — APPROVED** [review]

**Pull request merged**